AppKit/CoreAnimation/CPAnimationContext.j

```diff
@@ -152,24 +152,29 @@ CPLog.debug(_cmd + "context stack =" + _CPAnimationContextStack);
 
     duration = [animation duration] || [self duration];
 
-    var needsFrameTimer = (aKeyPath == @"frame" || aKeyPath == @"frameSize") &&
-                          ([anObject hasCustomLayoutSubviews] || [anObject hasCustomDrawRect]) &&
-                          (objectId = [anObject UID]);
+    var needsPeriodicFrameUpdates = (aKeyPath === @"frame" || aKeyPath === @"frameSize" || aKeyPath === @"frameOrigin") &&
+                                    [[anObject animator] wantPeriodicFrameUpdates] &&
```

Comment:
Because -layoutSubviews needs the receiver frameSize but not the frameOrigin. If you show me an example where the subviews are displayed based on the superview frameOrigin I will admit I'm wrong. As I said, frameOrigin updates should be opt-in. They are needed in your case because you need to draw from the superview but it's not the common case.